lib/rubocop/cop/gemspec/ruby_version_globals_usage.rb

```diff
@@ -0,0 +1,55 @@
+# frozen_string_literal: true
+
+module RuboCop
+  module Cop
+    module Gemspec
+      # Checks that `RUBY_VERSION` constant is not used in gemspec.
+      # Using `RUBY_VERSION` is dangerous because value of the
+      # constant is determined by `rake release`.
+      # It's possible to have dependency based on ruby version used
+      # to execute `rake release` and not user's ruby version.
+      #
+      # @example
+      #
+      #   # bad
+      #   Gem::Specification.new do |spec|
+      #     if RUBY_VERSION >= '2.5'
+      #       spec.add_runtime_dependency 'gem_a'
+      #     else
+      #       spec.add_runtime_dependency 'gem_b'
+      #     end
+      #   end
+      #
+      #   # good
+      #   Gem::Specification.new do |spec|
+      #     spec.add_runtime_dependency 'gem_a'
+      #   end
+      #
+      class RubyVersionGlobalsUsage < Cop
+        MSG = '`RUBY_VERSION` used in gemspec file.'
```

Comment:
The following may be a bit clearer as a warning.

```diff
- MSG = '`RUBY_VERSION` used in gemspec file.'
+ MSG = 'Do not use `RUBY_VERSION` in gemspec file.`
```

Reply:
Yep, sounds better. Fixed.